idaes/core/base/costing_base.py

```diff
@@ -345,22 +345,24 @@ def register_flow_type(self, flow_type, cost):
         self.flow_types.add(flow_type)
 
         name = f"{flow_type}_cost"
-        current_component = self.component(name)
+        current_component = getattr(self, name, None)
         if current_component is not None:
             if current_component is not cost:
                 raise RuntimeError(
                     f"Component {name} already exists on {self} but is not {cost}."
                 )
             # now self.{flow_type}_cost is cost, so just use it
         else:
-            # Create a Var to hold the cost
-            # Units will be different between flows, so have to use scalar Vars
-            fvar = pyo.Var(
-                units=pyo.units.get_units(cost),
-                doc=f"Cost parameter for {flow_type} flow",
-            )
-            self.add_component(name, fvar)
-            fvar.fix(cost)
+            add_object_reference(self, name, cost)
```

Comment:
The original intent was that `cost` would be a `float` which would be used to initialize the new `Var` - this change would break that usage as no `Var` would be created (the new `self.name` attribute would just be a pointer to the `float`).